addons/web_editor/static/src/xml/wysiwyg.xml

```diff
@@ -327,10 +327,43 @@
         </t>
     </t>
 
+    <t t-name="wysiwyg.widgets.video.existing.attachments">
+        <div class="o_we_existing_attachments o_we_videos">
+            <div t-if="!widget.hasContent()" class="o_nocontent_help">
+                <p class="o_empty_folder_video">No videos found.</p>
+                <p class="o_empty_folder_subtitle">You can upload videos with the button located in the top right of the screen.</p>
+            </div>
+            <div t-else="" class="row mx-auto">
+                <t t-foreach="attachments" t-as="attachment">
+                    <div class="col-2 o_existing_attachment_cell o_we_attachment_highlight my-2" t-att-data-id="attachment.id">
+                        <t t-call="wysiwyg.widgets.file.existing.remove"/>
+                        <t t-if="attachment.thumbnail">
+                            <div class="o_we_media_dialog_img_wrapper o_we_video_thumbnail">
+                                <img class="img img-fluid o_we_attachment_highlight" t-att-src="attachment.thumbnail" t-att-alt="attachment.name" t-att-title="attachment.name"/>
+                            </div>
+                        </t>
+                        <t t-else="">
+                            <div t-att-data-url="attachment.url" role="img" t-att-aria-label="attachment.name" t-att-title="attachment.name" t-att-data-mimetype="attachment.mimetype" t-attf-class="o_image d-flex align-items-center justify-content-center"/>
+                            <small class="o_file_name" t-esc="attachment.name"/>
+                            <img t-att-src="attachment.thumbnail"/>
+                        </t>
+                    </div>
+                </t>
+            </div>
+        </div>
+    </t>
+
     <!-- Video choosing part of the Media Dialog -->
     <t t-name="wysiwyg.widgets.video">
         <form action="#" class="row">
             <div class="col mt-4 o_video_dialog_form">
+                <div class="o_video_user_upload">
+                    <t t-call="wysiwyg.widgets.file">
+                        <t t-set="uploadText">Upload a video</t>
+                        <t t-set="searchPlaceholder">Search a video</t>
+                        <t t-set="urlWarningTitle">Uploaded video's format is not supported. File has to be .mp4 </t>
```

Comment:
The term "video format" is common, this looks strange to me. But we cannot say "video's video format" either.
What about saying "Format of the uploaded video..." instead ?